05_ANALYSIS_TOOLS/etl_pipeline/pipeline_runner.py

```
@@ -0,0 +1,361 @@
+            logger.info("Phase 3: Data Loading")
+            if not self._load_data():
+                return False
+            
+            # Pipeline completed successfully
+            self.pipeline_metadata['end_time'] = datetime.now()
+            self.pipeline_metadata['status'] = 'completed'
+            
+            duration = self.pipeline_metadata['end_time'] - self.pipeline_metadata['start_time']
+            logger.info(f"ETL pipeline completed successfully in {duration}")
+            
+            return True
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)